[Cloud Security] [Alerts] Workflow enhancements for Alerts in Cloud Security

opauloh opened the pull request:
## Summary

It closes #164793

This PR adds a few enhancements to the Alerts workflow in Cloud Security:

- Changing wording `x alerts detected by y rules` to `x alerts detected by y detection rules`
- Adding investigation fields for rules created from findings and vulnerabilities 
- Adding investigation guide for rules created from findings

### Screenshot


Text changes

<img width="1501" alt="image" src="https://github.com/elastic/kibana/assets/19270322/cad550dd-bcba-4487-a256-c09b7769b437">

Highlighted fields in the rule

<img width="1475" alt="image" src="https://github.com/elastic/kibana/assets/19270322/5d7fcb52-d3c5-4b7f-9c1f-d1c4a773dcc1">

investigation guide in rule

<img width="1484" alt="image" src="https://github.com/elastic/kibana/assets/19270322/5ca0042f-d632-46bb-a6c7-22d75440e869">

Highlighted fields in the alert

<img width="1486" alt="image" src="https://github.com/elastic/kibana/assets/19270322/1fe464ad-77f8-4a25-baa2-05af5a260058">

investigation guide in alert

<img width="1113" alt="image" src="https://github.com/elastic/kibana/assets/19270322/2864c9c5-cb76-4184-9231-f8ffcf20c119">

elasticmachine commented:
Pinging @elastic/kibana-cloud-security-posture (Team:Cloud Security)

JordanSh commented:
![image](https://github.com/elastic/kibana/assets/51442161/9f366a15-b69a-43c8-9c85-404060a71a25)

My call fails because its missing a version, is this something that is expected to happen locally?
Besides that, the Take Action button is stuck in loading state even though the call already failed, we should probably fix this as well, should be simple enough.

Maybe also add error handling to quick wins? @kfirpeled

maxcold reviewed (commented):
Great low-hanging fruits for the UX! The code looks good, but let's rebase first with the api version PR so it's possible to test locally properly without patching the calls

maxcold commented:
@opauloh Attempted to test the PR again and getting 
```
error: "Bad Request"
message: "[request body]: Invalid value \"[\"resource.name\",\"resource.id\",\"resource.type\",\"resource.sub_type\"]\" supplied to \"investigation_fields\""
statusCode: 400
```
now on rule creation. Same error in vulnerabilities (the list of values is different ). Not sure what's off there

jfyi I also rebased the PR as initially I thought it's the same error with version

opauloh commented:
> @opauloh Attempted to test the PR again and getting
> 
> ```
> error: "Bad Request"
> message: "[request body]: Invalid value \"[\"resource.name\",\"resource.id\",\"resource.type\",\"resource.sub_type\"]\" supplied to \"investigation_fields\""
> statusCode: 400
> ```
> 
> now on rule creation. Same error in vulnerabilities (the list of values is different ). Not sure what's off there
> 
> jfyi I also rebased the PR as initially I thought it's the same error with version

Thank you for reporting this, the API has changed from the moment of the pull request to the review, I just fixed it with the updated field, can you test it again?

On a side note, I'm working now on setting up FTR for alerts, so we will be able to catch breaking changes like this, it will be a separate PR but I'll cover this PR changes as well when it's ready

maxcold reviewed (approved):
Tested, rules are being created correctly, with the remediation and highlighted fields as expected.
